### ISSUE TITLE
Wrong positioning of the search marks

### ISSUE THREAD

**Issue by kenneth488** [KurtCode/PDFKitten#85]
Hi KurtCode

I have the library with different PDF files, and it does often place the search marks wrong. In my fork, there is a PDF file called amazon-dynamo-sosp2007.pdf. Searching for the string "re" will show you that the position of the marks a sometimes calculated wrong, since they are misplaced. The same goes for the string "in" and probably others.

Is this something you might be able to fix?

Have a great day

Regards,

Kenneth

**Comment by viezel** [comment]
+1 could be nice to get sorted out.

**Comment by kenneth488** [comment]
I have almost fixed the issue now, so the general solution works. Will commit and push today.

**Comment by TryMyBest** [comment]
Hello. I really really need your new version. Plz commit it.

**Comment by kenneth488** [comment]
I will commit it as soon as all the bugs are fixed! Are working on it atm! :)

**Comment by TryMyBest** [comment]
Mate! I fixed this bug. Do you have skype or contact. I think i have some ideas for this project like search multi-word. And done! Add me and i can show you:).

**Comment by TryMyBest** [comment]
Hey mate. Are you there? I think i can modify this project to search multi-word :)

**Comment by kenneth488** [comment]
I will commit the fixes soon. I also implemented support for newline word searching, so we can mark search words spanning more than one line. 

When I commit, you can fork it and implement your fixes.

**Comment by TryMyBest** [comment]
Ok. Will check your new version. Nice work guy! :)

**Comment by nattapon21** [comment]
.....


2015-06-12 16:18 GMT+07:00 TryMyBest <notifications@github.com>:

**Comment by kenneth488** [comment]
The commit is up. Now remember that there might be more bugs. If so post them. I am still working on making it a lot better.

**Comment by TryMyBest** [comment]
1. When we zoom out and search, draw is wrong. Old version does not.
2. Same when search words on two lines.

**Comment by kenneth488** [comment]
Yes just saw that. Though that is only in the test project. It does work when you give it the correct page size. The way Kitten sets up a page, is pretty weird. It works fine otherwise:
http://postimg.org/image/f5py9l2d7/
http://postimg.org/image/ksmb724vv/

I will see if I can fix it in Kitten.

**Comment by kenneth488** [comment]
What you need to pass to the convertPDFRectToViewRect method is the frame of the userSpace PDF page aka. the visible pdf page you are viewing. Anything else will fail.

**Comment by TryMyBest** [comment]
Not understand what you said :D.

**Comment by TryMyBest** [comment]
With me, that is not problem because when searching, the pages are invisible so they cannot zoom:).

**Comment by timeflying** [comment]
Still mark on the wrong position~

http://www.takming.edu.tw/study/upload/filedownload150318163116.pdf

![simulator screen shot aug 1 2016 3 47 00 pm](https://cloud.githubusercontent.com/assets/6119146/17286930/3bda5d52-57ff-11e6-8658-b06749c4b01c.png)

**Comment by timeflying** [comment]
I use the app preview which  embedded in mac os . It works great on the same pdf.
![screen shot 2016-08-01 at 3 52 49 pm](https://cloud.githubusercontent.com/assets/6119146/17287115/4c554588-5800-11e6-955f-b364bfdcf410.png)
